compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -250,26 +245,31 @@ static VectorPreProcStrategy getVectorPreProcStrategy(
 
 /// Looks for the `native_vector_size` attribute in the hal.executable.target
 /// looked up from this op.
-static std::optional<int64_t> getNativeVectorSizeInBytes(
-    func::FuncOp entryPointFn) {
+static int64_t getNativeVectorSizeInBytes(func::FuncOp entryPointFn) {
   auto targetAttr = IREE::HAL::ExecutableTargetAttr::lookup(entryPointFn);
   auto nativeVectorSizeAttr =
       getConfigIntegerAttr(targetAttr, "native_vector_size");
-  if (!nativeVectorSizeAttr) return std::nullopt;
-  int64_t nativeVectorSizeVal = nativeVectorSizeAttr->getInt();
-  if (!nativeVectorSizeVal) return std::nullopt;
-  return nativeVectorSizeVal;
+  if (nativeVectorSizeAttr) {
+    int64_t nativeVectorSizeVal = nativeVectorSizeAttr->getInt();
+    if (nativeVectorSizeVal) {
+      return nativeVectorSizeVal;
+    }
+  }
+
+  // TODO(dcaballe): Remove this workaround for VMVX.
+  if (isVMVXBackend(targetAttr)) {
+    constexpr int64_t defaultNativeVectorSizeforVMVX = 16;
+    return defaultNativeVectorSizeforVMVX;
+  }
+
+  llvm_unreachable("Missing 'native_vector_size' attribute");
```

Comment:
I think the convention is to now use `llvm_unreachable`... (Benoit can fill you in on all the details).. Convert to`assert` or best return a `Failure` and catch appropriately.